lib/net/ssh/authentication/ed25519.rb

```diff
@@ -19,18 +10,25 @@
 
 module Net; module SSH; module Authentication
 module ED25519
-  class SigningKeyFromFile < RbNaCl::Signatures::Ed25519::SigningKey
+  class SigningKeyFromFile < SimpleDelegator
     def initialize(pk,sk)
-      @signing_key = sk
-      @verify_key = RbNaCl::Signatures::Ed25519::VerifyKey.new(pk)
+      key = ::Ed25519::SigningKey.from_keypair(sk)
+      raise ArgumentError, "pk does not match sk" unless pk == key.verify_key.to_bytes
+
+      super(key)
+    end
+
+    # Deprecated method provided by the RbNaCl API
+    def keypair_bytes
```

Comment:
This should be the only method available in the RbNaCl implementation which is not available in the ed25519 gem. I went ahead and put this here for compatibility but changed all of the usages of it throughout the gem to the new API (which is just `#keypair` and is an attr reader)